cvat/apps/engine/views.py

```diff
@@ -156,7 +157,8 @@ def share(request):
                         "mime_type": entry_mime_type,
                     })
 
-            serializer = FileInfoSerializer(many=True, data=data)
+            # return directories at the top of the list
+            serializer = FileInfoSerializer(many=True, data=sorted(sorted(data, key=lambda x: x['name']), key=lambda x: x['type']))
```

Comment:
Can this be sorted by (x['name'], x['type']) tuple key?

Reply:
Thanks, updated to (x['type'], x['name'])